Automattic/wp-calypso
# Components: Remove Accordion and replace it with Panel

After some recent [Calypso editor cleanup](https://github.com/Automattic/wp-calypso/pull/47095) it turns out that the Calypso `<Accordion />` component is used only at a couple of locations in the Hosting settings screen.

That makes it a good and simple enough candidate to address in our quest to make `@wordpress/components` the canonical component library throughout all our applications and interfaces.

This PR updates those accordions to use a `PanelBody` from `@wordpress/components` and removes the `<Accordion />` component completely.

It is immediately visible that the new component is more accessible and does a better job to indicate user focus.

Furthermore, the arrows of `Panel` are much more in sync with other UI elements in Calypso, like the sidebar menu accordions.

In my opinion, this is an easy win and a great example of how simple and beneficial a migration to a `@wordpress/components` component can be.

There are some design changes here, so I'd appreciate feedback from the design team.

#### Changes proposed in this Pull Request

* Components: Remove Accordion and replace it with Panel

#### Preview

Default state, collapsed - before:
![](https://cldup.com/Z212JuinHL.png)

Default state, collapsed - after:
![](https://cldup.com/DiZJP1tGLp.png)

Hover state, collapsed - before:
![](https://cldup.com/g1_20Qet9a.png)

Hover state, collapsed - after:
![](https://cldup.com/eUFm_GV-47.png)

Focused state, collapsed - before (doesn't have any focus indication):
![](https://cldup.com/Z212JuinHL.png)

Focused state, collapsed - after:
![](https://cldup.com/8A-6OGEpCX.png)

Default state, expanded - before:
![](https://cldup.com/WJPqorHV58.png)

Default state, expanded - after:
![](https://cldup.com/cqwKFlOuZn.png)


#### Testing instructions

* Go to `/hosting-config/:site` where `:site` is the slug of an Atomic site without SFTP credentials.
* Verify the SFTP credentials accordion looks and works well.
* Verify the phpmyadmin accordion looks and works well.
